pinot-server/src/main/java/org/apache/pinot/server/starter/helix/HelixInstanceDataManager.java

```diff
@@ -509,6 +544,44 @@ public void addOrReplaceSegment(String tableNameWithType, String segmentName)
     }
   }
 
+  @Override
+  public void reloadTable(String tableNameWithType, boolean force, SegmentRefreshSemaphore segmentRefreshSemaphore) {
+    TableConfig tableConfig = ZKMetadataProvider.getTableConfig(_propertyStore, tableNameWithType);
+    AtomicReference<TableDataManager> toBeShutdown = new AtomicReference<>();
+    _tableDataManagerMap.computeIfPresent(tableNameWithType, (k, tdm) -> {
+      // create only if a table data manager doesn't exist
+      // or the current configuration and the new configuration are different
+      if (!force && tdm.getTableDataManagerConfig().getTableConfig().equals(tableConfig)) {
```

Comment:
We can compare the ZNode version of the table config to identify if it is changed